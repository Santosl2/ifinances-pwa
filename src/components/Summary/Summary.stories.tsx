import { ComponentMeta, ComponentStory } from "@storybook/react";

import { Summary } from "./Summary";

export default {
  title: "Components/Summary",
  component: Summary,
  args: {
    data: {
      total: {
        income: 5000,
        outcome: 200,
        final: 4800,
      },
    },
  },
} as ComponentMeta<typeof Summary>;

const Template: ComponentStory<typeof Summary> = (args) => {
  return (
    <div style={{ marginTop: "12rem" }}>
      <Summary {...args} />
    </div>
  );
};

export const Default = Template.bind({});

Default.args = {
  isLoading: false,
};
