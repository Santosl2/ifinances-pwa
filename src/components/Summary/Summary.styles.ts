import styled from "styled-components";

export const SummaryWrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  gap: 1.5rem;
  margin-top: -9rem;
  margin-bottom: 5rem;
  position: relative;
  z-index: 1;

  div:last-child {
    background-color: var(--green);
    color: var(--shape);
  }
`;
