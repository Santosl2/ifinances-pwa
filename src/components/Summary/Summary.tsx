import { Order } from "./Order";
import { OrderIcons } from "./Order/Order.types";
import { SummaryWrapper } from "./Summary.styles";
import { SummaryProps } from "./Summary.types";

export function Summary({ data, isLoading }: SummaryProps): JSX.Element {
  return (
    <SummaryWrapper>
      <Order
        title="Entradas"
        amount={data?.total?.income || 0}
        icon={OrderIcons.income}
        isLoading={isLoading}
      />
      <Order
        title="SaÃ­das"
        amount={data?.total?.outcome || 0}
        icon={OrderIcons.outcome}
        isLoading={isLoading}
      />
      <Order
        title="Total"
        amount={data?.total?.final || 0}
        icon={OrderIcons.total}
        isLoading={isLoading}
      />
    </SummaryWrapper>
  );
}
