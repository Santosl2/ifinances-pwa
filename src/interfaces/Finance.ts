// eslint-disable-next-line no-shadow
export enum FinanceTypes {
  "income" = "income",
  "outcome" = "outcome",
  "total" = "total",
}
