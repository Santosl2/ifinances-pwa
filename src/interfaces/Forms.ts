export type SignUpFormData = {
  name: string;
  email: string;
  password: string;
  password_confirmation: string;
};

export type SignInFormData = {
  email: string;
  password: string;
};
