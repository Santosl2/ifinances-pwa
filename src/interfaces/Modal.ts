export type ModalProps = {
  isOpen: boolean;
};

export type ModalSelector = {
  modal: ModalProps;
};
