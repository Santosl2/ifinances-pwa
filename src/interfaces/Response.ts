import { FinanceTypes } from "./Finance";

export type UserFinancesResponse = {
  amount: number;
  type: FinanceTypes;
  userId: string;
  date: string;
};
