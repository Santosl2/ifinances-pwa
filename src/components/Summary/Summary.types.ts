import { UserFinancesResponse } from "@/interfaces/Response";

export interface SummaryProps {
  data: UserFinancesResponse | undefined;
  isLoading: boolean;
}
