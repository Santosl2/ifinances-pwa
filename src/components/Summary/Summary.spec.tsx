/* eslint-disable testing-library/no-node-access */
/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable unused-imports/no-unused-imports */
import * as React from "react";

import { UserFinancesResponse } from "@/interfaces/Response";
import { render, screen } from "@testing-library/react";
import "jest-styled-components";

import { Summary } from "./Summary";

let data: UserFinancesResponse;

describe("Summary test", () => {
  beforeEach(() => {
    data = {
      data: [],
      total: {
        income: 0,
        outcome: 0,
        final: 0,
      },
    };
  });

  it("should render summary", async () => {
    render(<Summary data={data} isLoading={false} />);

    expect(screen.getByText("Entradas")).toBeInTheDocument();
    expect(screen.getByText("SaÃ­das")).toBeInTheDocument();
    expect(screen.getByText("Total")).toBeInTheDocument();
  });

  it("should br match snapshot", async () => {
    const { container } = render(<Summary data={data} isLoading={false} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
