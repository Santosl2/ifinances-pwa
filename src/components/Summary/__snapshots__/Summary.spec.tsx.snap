// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary test should br match snapshot 1`] = `
.c1 {
  padding: 1.2rem 1.5rem;
  border-radius: 0.2rem;
  background: var(--shape);
  color: var(--text-title);
  box-shadow: 0 0 5px 0 rgba(0,0,0,0.2);
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  cursor: pointer;
}

.c1:hover {
  -webkit-transform: translateY(-1rem);
  -ms-transform: translateY(-1rem);
  transform: translateY(-1rem);
}

.c1 strong {
  display: block;
  margin-top: 1rem;
  font-size: 2rem;
  font-weight: 500;
  line-height: 3rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 img {
  margin-left: auto;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(250px,1fr));
  gap: 1.5rem;
  margin-top: -9rem;
  margin-bottom: 5rem;
  position: relative;
  z-index: 1;
}

.c0 div:last-child {
  background-color: var(--green);
  color: var(--shape);
}

<div
  class="c0"
  data-testid="summaryTest"
>
  <div
    class="c1"
    style="opacity: 0; transform: translateY(50%) translateZ(0);"
  >
    <header
      class="c2"
    >
      <p>
        Entradas
      </p>
      <img
        alt="Imagem Order"
        src="income.svg"
      />
    </header>
    <strong>
      R$
      <span
        data-amount="0"
      >
        0
      </span>
    </strong>
  </div>
  <div
    class="c1"
    style="opacity: 0; transform: translateY(50%) translateZ(0);"
  >
    <header
      class="c2"
    >
      <p>
        Sa√≠das
      </p>
      <img
        alt="Imagem Order"
        src="outcome.svg"
      />
    </header>
    <strong>
      R$
      <span
        data-amount="0"
      >
        0
      </span>
    </strong>
  </div>
  <div
    class="c1"
    style="opacity: 0; transform: translateY(50%) translateZ(0);"
  >
    <header
      class="c2"
    >
      <p>
        Total
      </p>
      <img
        alt="Imagem Order"
        src="total.svg"
      />
    </header>
    <strong>
      R$
      <span
        data-amount="0"
      >
        0
      </span>
    </strong>
  </div>
</div>
`;
